File: memory-garden-server/src/main/java/com/memorygarden/model/entity/Category.java
package com.memorygarden.model.entity;

import lombok.Data;

import java.util.Date;

/**
 * 分类实体
 *
 * @author jLU
 * @date 2026-05-20
 */
@Data
public class Category {

    /**
     * 主键
     */
    private Long id;

    /**
     * 所属用户
     */
    private Long userId;

    /**
     * 分类名称
     */
    private String name;

    /**
     * 分类图标标识
     */
    private String icon;

    /**
     * 排序序号
     */
    private Integer sortOrder;

    /**
     * 软删除：0-未删除，1-已删除
     */
    private Integer isDeleted;

    /**
     * 创建时间
     */
    private Date createTime;

    /**
     * 更新时间
     */
    private Date updateTime;
}

File: memory-garden-web/src/api/category.ts
import { post, get, put, del } from './request'

/**
 * 分类实体
 */
export interface Category {
  id: number
  userId: number
  name: string
  icon: string
  sortOrder: number
  isDeleted: number
  createTime: string
  updateTime: string
}

/**
 * 分类 API 封装
 */
export const categoryApi = {
  /**
   * 创建分类
   */
  create(data: { name: string; icon?: string; sortOrder?: number }) {
    return post<number>('/category', data)
  },

  /**
   * 获取分类列表
   */
  list() {
    return get<Category[]>('/category/list')
  },

  /**
   * 修改分类
   */
  update(id: number, data: { name?: string; icon?: string; sortOrder?: number }) {
    return put<boolean>(`/category/${id}`, data)
  },

  /**
   * 删除分类
   */
  delete(id: number) {
    return del<boolean>(`/category/${id}`)
  }
}
